src/StorageOptions.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace PodPoint\MailExport;
+
+use Carbon\Carbon;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Str;
+use RuntimeException;
+use Swift_Message;
+
+/**
+ * Data transfer object responsible for holding
+ * storage informations when exporting a mail.
+ */
+class StorageOptions
+{
+    const EXTENSION = 'eml';
+
+    /**
+     * @var string
+     */
+    public $disk;
+
+    /**
+     * @var string
+     */
+    public $path;
+
+    /**
+     * @var string
+     */
+    public $filename;
+
+    /**
+     * @var \Swift_Message
+     */
+    public $message;
+
+    public function __construct(Swift_Message $message, array $parameters = [])
+    {
+        $this->message = $message;
+
+        $parameters = Arr::only($parameters, ['disk', 'path', 'filename']);
+
+        foreach ($parameters as $propertyName => $propertyValue) {
+            $default = 'default'.Str::studly($propertyName);
+            $this->$propertyName = $propertyValue ?: $this->$default();
+        }
+    }
+
+    public function __set($name, $value)
```

Comment:
I'll remove it because:
- well first of all I'm leveraging the dynamic language to do this, feels a bit ironic to try to add some strictness back in
- I could actually see myself setting it on the fly if needed, don't see why not...